http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcClientContext.java

```diff
@@ -94,10 +94,10 @@ protected OidcClientConfiguration resolveUrls(OidcClientConfiguration deployment
         } else {
             OidcClientConfigurationDelegate delegate = new OidcClientConfigurationDelegate(this.oidcClientConfig);
             if (oidcClientConfig.getAuthServerBaseUrl() != null) {
-                delegate.setAuthServerBaseUrl(getAuthServerBaseUrl(facade, this.oidcClientConfig.getAuthServerBaseUrl()));
+                delegate.setAuthServerBaseUrl(getBaseUrlOfIdentityProvider(facade, this.oidcClientConfig.getAuthServerBaseUrl()));
             }
             if (oidcClientConfig.getProviderUrl() != null) {
-                delegate.setProviderUrl(oidcClientConfig.getProviderUrl());
```

Comment:
The `providerUrl` is meant to be a generic configuration option for specifying the provider URL for any OpenID provider. The ability to specify a relative URL seems specific to when the OpenID provider is Keycloak. If that's the case, I don't think it makes sense to update `providerUrl` to support relative URLs in general. For Keycloak-specific configuration, `authServerUrl` can continue to be used.